api/search.go

```diff
@@ -117,6 +118,33 @@ func parseURIPrefix(ctx context.Context, params url.Values) (uriPrefix string, e
 	return uriPrefix, nil
 }
 
+// validateCDIDs checks that all the provided CDIDs are not blank and 4 or 5 characters long
+// returns nil and an empty array if all of them are valid,
+// returns error and a list of CDIDs that are not valid, if at least one is not valid
+func validateCDIDs(cdids []string) (invalidCDIDs []string, err error) {
+	if len(cdids) == 0 {
+		return nil, nil
+	}
+
+	validationRegex := regexp.MustCompile(`^[a-zA-Z0-9]{4,5}$`)
+
+	for _, cdid := range cdids {
+		if !validationRegex.MatchString(cdid) {
+			if cdid == "" {
+				invalidCDIDs = append(invalidCDIDs, "<blank>")
+			} else {
+				invalidCDIDs = append(invalidCDIDs, cdid)
+			}
+		}
+	}
+
+	if len(invalidCDIDs) > 0 {
+		err = fmt.Errorf("invalid cdid(s) found")
```

Comment:
```suggestion
		err = errors.New("invalid cdid(s) found")
```
since it's no longer formatted